extensions/resteasy-reactive/quarkus-resteasy-reactive-jackson/runtime/src/main/java/io/quarkus/resteasy/reactive/jackson/runtime/serialisers/ServerJacksonMessageBodyReader.java

```diff
@@ -53,10 +54,22 @@ public Object readFrom(Class<Object> type, Type genericType, MediaType mediaType
     }
 
     private Object doReadFrom(Class<Object> type, Type genericType, InputStream entityStream) throws IOException {
-        if (entityStream instanceof EmptyInputStream) {
+        if (StreamUtil.isEmpty(entityStream)) {
             return null;
         }
-        return reader.forType(reader.getTypeFactory().constructType(genericType != null ? genericType : type))
-                .readValue(entityStream);
+        try {
+            return reader.forType(reader.getTypeFactory().constructType(genericType != null ? genericType : type))
+                    .readValue(entityStream);
+        } catch (MismatchedInputException e) {
+            if (isEmptyInputException(e)) {
+                return null;
+            }
+            throw e;
+        }
+    }
+
+    private boolean isEmptyInputException(MismatchedInputException e) {
+        // this isn't great, but Jackson doesn't have a specific exception for empty input...
+        return e.getMessage().startsWith("No content");
```

Comment:
🤣

Reply:
The difficlties of dealing with the real world...